### ISSUE TITLE
Add Page title in Hubspot Form integration context data

### ISSUE THREAD

**Issue by umkasanki** [verbb/formie#2029]
### What are you trying to do?

If I try to submit a HubSpot form via the Formie integration, I only see the url in the results table. It looks like the form page title is not being added to the context when the form is submitted.
![изображение](https://github.com/user-attachments/assets/627158c7-b706-4efb-9c07-c9835e9df6e6)

### What's your proposed solution?

Could you please add the form page title to the HubSpot context as provided in the Forms API??
```
  "context": {
    "hutk":"hutk", // include this parameter and set it to the hubspotutk cookie value to enable cookie tracking on your submission
    "pageUri": "www.example.com/page",
    "pageName": "Example page"
  },
```
https://legacydocs.hubspot.com/docs/methods/forms/submit_form_v3_authentication#:~:text=As%20this%20API%20is%20authenticated%2C

Result:
![изображение](https://github.com/user-attachments/assets/3311e487-3a0f-4cf9-b94f-4239b3e22a3f)

Thanks

**Comment by engram-design** [comment]
We already automatically include the [`pageUri`](https://github.com/verbb/formie/blob/877095faacca2aa87f83060e19bd0476cb632e67/src/integrations/crm/HubSpot.php#L444) but as for the `pageName` that's a little trickier. 

It's not something we can map in the form integration settings without having it map to a Hidden field that you'd populate with the entry's title when rendering the form. 

We could _maybe_ scrape it from the referrer's meta title, but I'd rather not add that sort of overhead.

In Formie 3 for Craft 5, you'll be able to use PHP events in a module to add this sort of thing as context variables for the integration, to send along with the payload.

For now, I've added "Page Name" and "Page URI" to the mapping interface where you can handle all this through hidden fields, and populate them in your templates.

Updated for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.26"`.

**Comment by umkasanki** [comment]
> For now, I've added "Page Name" and "Page URI" to the mapping interface where you can handle all this through hidden fields, and populate them in your templates.

Excellent. Thank you very much!

**Comment by umkasanki** [comment]
Hello @engram-design 

Checked. Works.
But with this implementation the client gets the following warning in the HubSpot account.
`This value wasn't passed to the Contact record. 'pagename' isn't a Contact property.`

![изображение](https://github.com/user-attachments/assets/a4aa18dd-b463-49b2-a419-b1c9d8ff5daa)


Looks a bit dirty.
Maybe there is another, more beautiful way to pass the form page title to the context?

**Comment by engram-design** [comment]
Ah, that's my mistake not removing them from the mapping of the contact object. Updated for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.26"`.

**Comment by umkasanki** [comment]
I still get this warning((.

**HubSpot submission:**
![изображение](https://github.com/user-attachments/assets/74c30292-a239-4130-b8ab-cd1c1cf2ccb6)

**Form mapping:**
![изображение](https://github.com/user-attachments/assets/d9863b54-5c84-43ca-9291-bcf480590ab6)

**Update is installed:**
![изображение](https://github.com/user-attachments/assets/fd282d35-453e-49af-841f-56e729fa6cbb)

**Comment by engram-design** [comment]
Should have double-checked that, sorry. Updated for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.26"`.

**Comment by engram-design** [comment]
Fixed in [2.1.27](https://github.com/verbb/formie/releases/tag/2.1.27)
